### PR TITLE
Always prefill username from Windows login name based on server version

### PR DESCRIPTION
**Pull request by allexzander**
Signed-off-by: alex-z <blackslayer4@gmail.com>

### CODE REVIEW

**Comment by allexzander** [comment]
As discussed with @tobiasKaminsky we want this enabled by default starting from the first server release (24 according to this comment https://github.com/nextcloud/server/pull/29530#issuecomment-958985268) which supports the `user` query parameter.

**Comment by allexzander** [comment]
@mgallien See my comment explaining this change

File: VERSION.cmake [hunk 1]
```diff
@@ -9,6 +9,8 @@ set(NEXTCLOUD_SERVER_VERSION_MIN_SUPPORTED_MAJOR 16)
 set(NEXTCLOUD_SERVER_VERSION_MIN_SUPPORTED_MINOR 0)
 set(NEXTCLOUD_SERVER_VERSION_MIN_SUPPORTED_PATCH 0)
 
+set(NEXTCLOUD_USERNAME_PREFILL_SERVER_VERSION_MIN_SUPPORTED_MAJOR 24)
```
Comment (mgallien) [inline VERSION.cmake:12]: why not just a constant in the code ?

**Review by mgallien — APPROVED** [review]
looks good

**Review by camilasan — APPROVED** [review]

**Pull request merged**